src/components/structures/RoomView.js

```diff
@@ -615,33 +615,17 @@ module.exports = React.createClass({
                 continue;
             }
 
-            var eventId = mxEv.getId();
-
             if (this.state.searchScope === 'All') {
                 var roomId = mxEv.getRoomId();
                 if(roomId != lastRoomId) {
-                    ret.push(<li key={eventId + "-room"}><h1>Room: { cli.getRoom(roomId).name }</h1></li>);
+                    ret.push(<li key={mxEv.getId() + "-room"}><h1>Room: { cli.getRoom(roomId).name }</h1></li>);
```

Comment:
We should be null guarding on `getRoom(roomId)` - We're grabbing the ID from the event returned from a search request but there are no guarantees that the `MatrixClient` will be aware of it (from `/sync`). This is mainly due to archived rooms which the JS SDK will lazily fetch, but will (should) still be returned from searches.